src/wp-includes/pluggable.php

```diff
@@ -444,7 +444,13 @@ function wp_mail( $to, $subject, $message, $headers = '', $attachments = array()
 		$from_name = apply_filters( 'wp_mail_from_name', $from_name );
 
 		try {
-			$phpmailer->setFrom( $from_email, $from_name );
+			// This block should be removed once PHPMailer supports setting the envelope sender separately.
+			$sendmail_path = ini_get( 'sendmail_path' );
+			if ( str_contains( $sendmail_path, '-f' ) ) {
+				$phpmailer->setFrom( $from_email, $from_name, false );
+			} else {
+				$phpmailer->setFrom( $from_email, $from_name );
+			}
```

Comment:
This would be more concise:

```suggestion
			$auto = ! str_contains( $sendmail_path, '-f' );
			$phpmailer->setFrom( $from_email, $from_name, $auto );
```